packages/stems/src/components/HarmonyButton/HarmonyButton.tsx

```diff
@@ -0,0 +1,105 @@
+import { CSSProperties, forwardRef } from 'react'
+
+import cn from 'classnames'
+
+import { useMediaQueryListener } from 'hooks/useMediaQueryListener'
+import { CSSCustomProperties } from 'styles/types'
+import { toCSSVariableName } from 'utils/styles'
+
+import styles from './HarmonyButton.module.css'
+import {
+  HarmonyButtonProps,
+  HarmonyButtonType,
+  HarmonyButtonSize
+} from './types'
+
+const SIZE_STYLE_MAP: { [k in HarmonyButtonSize]: string } = {
+  [HarmonyButtonSize.SMALL]: styles.small,
+  [HarmonyButtonSize.DEFAULT]: styles.default,
+  [HarmonyButtonSize.LARGE]: styles.large
+}
+
+const TYPE_STYLE_MAP: { [k in HarmonyButtonType]: string } = {
+  [HarmonyButtonType.PRIMARY]: styles.primary,
+  [HarmonyButtonType.SECONDARY]: styles.secondary,
+  [HarmonyButtonType.TERTIARY]: styles.tertiary,
+  [HarmonyButtonType.DESTRUCTIVE]: styles.destructive,
+  [HarmonyButtonType.GHOST]: styles.ghost
+}
+
+/**
+ * A common Button component. Includes a few variants and options to
+ * include and position icons.
+ */
+export const HarmonyButton = forwardRef<HTMLButtonElement, HarmonyButtonProps>(
+  function Button(
+    {
```

Comment:
just since this will be our "core components" ill be extra nitty just to make sure we have alignment. so i personally prefer `function Component(props) {...}` and then destructure props in the body. i find this makes the top-level function definition a but easier to parse. if you feel strongly this way, all good as well, just curious :)

Reply:
i forget but i think it also may prevent an extra nested body definition.

Reply:
I actually don't find either way more or less readable. Happy to destructure in the function body. I thought there was a style guide entry about doing it in the arguments, but I was unable to find it.

Reply:
yeah its not official, just something i personally like and wanted to start a convo :)